contracts/Swap/SwapRouter.sol

```diff
@@ -0,0 +1,312 @@
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
+     * @dev In case of swapping native BNB the first asset in path array should be the wBNB address
+     */
+    function swapBnbAndSupply(
+        address vTokenAddress,
+        uint256 amountOutMin,
+        address[] calldata path,
+        uint256 deadline
+    ) external payable override ensure(deadline) {
+        uint256[] memory swapAmounts = _swapExactETHForTokens(amountOutMin, path, msg.sender);
+        TransferHelper.safeApprove(path[1], vTokenAddress, 0);
+        TransferHelper.safeApprove(path[1], vTokenAddress, swapAmounts[1]);
+        uint256 response = IVToken(vTokenAddress).mintBehalf(msg.sender, swapAmounts[1]);
+        if (response != 0) {
+            revert SupplyError(msg.sender, vTokenAddress, response);
+        } else {
+            emit SupplyOnBehalf(msg.sender, vTokenAddress, swapAmounts[1]);
+        }
+    }
+
+    /**
+     * @notice Swap token A for token B and repays a borrow from a Venus Market
+     * @param vTokenAddress The address of the vToken contract to repay from.
+     * @param amountIn The amount of tokens to swap.
+     * @param amountOutMin Minimum amount of tokens to receive.
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
+     */
+    function swapAndRepay(
+        address vTokenAddress,
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        uint256 deadline
+    ) external override ensure(deadline) {
+        uint256[] memory swapAmounts = _swapExactTokensForTokens(amountIn, amountOutMin, path, msg.sender);
+        TransferHelper.safeApprove(path[1], vTokenAddress, 0);
+        TransferHelper.safeApprove(path[1], vTokenAddress, swapAmounts[1]);
+        uint256 response = IVToken(vTokenAddress).repayBorrowBehalf(msg.sender, swapAmounts[1]);
+        if (response != 0) {
+            revert RepayError(msg.sender, vTokenAddress, response);
+        } else {
+            emit RepayOnBehalf(msg.sender, vTokenAddress, swapAmounts[1]);
+        }
+    }
+
+    /**
+     * @notice Swap BNB for another token and repays a borrow from a Venus Market
+     * @dev The amount to be swapped is obtained from the msg.value, since we are swapping BNB
+     * @param vTokenAddress The address of the vToken contract to repay from.
+     * @param amountOutMin Minimum amount of tokens to receive.
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
```

Comment:
Could this array contain more than two assets? Is it the last asset that we are repaying?

Reply:
We should clarify that in the comment
```suggestion
     * @param path Array with addresses of the underlying assets to be swapped
     * @dev Addresses of underlying assets should be ordered that tokens to swap are listed first  and last asset is the token we receive
```